staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1573,22 +1574,49 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 }
 
 // startObservingCount starts monitoring given prefix and periodically updating metrics. It returns a function to stop collection.
-func (e *Store) startObservingCount(period time.Duration, objectCountTracker flowcontrolrequest.StorageObjectCountTracker) func() {
+func (e *Store) startObservingCount(period time.Duration, objectCountTracker flowcontrolrequest.StorageObjectCountTracker, isNamespaced bool) func() {
```

Comment:
I'm worried about this PR. We've seen clusters with thousands if not tens of thousands of namespaces.
If we multiple it by the number of resource types, that gives tremendous amount of metrics that we will be having and will need to periodically compute.

I agree that it would allow us to better estimate cost of list requests, but do we really have a convincing data that shows that it's really needed? It's not a reliability issue, the potential cost is making apiserver under-utilized, and I'm leaning towards this actually being a better tradeoff here.

Reply:
> I'm worried about this PR. We've seen clusters with thousands if not tens of thousands of namespaces.
If we multiple it by the number of resource types, that gives tremendous amount of metrics that we will be having and will need to periodically compute.

Fwiw we are not exposing the namespace level in the metric, but we are still tracking in-memory. I didn't think this is a huge cost though, because we are only tracking count per namespace, but I haven't done actual measurement on how much this can grow. 

> I agree that it would allow us to better estimate cost of list requests, but do we really have a convincing data that shows that it's really needed? It's not a reliability issue, the potential cost is making apiserver under-utilized, and I'm leaning towards this actually being a better tradeoff here.

I don't have a lot of data that indicate this is a big problem, but I have saw a case yesterday where a LIST request in a namespace with 1 object was costing 10 seats because the cluster overall had many objects of that resource. The "under-utilized" part is not a concern but if it results in other requests being rejected then it seems more problematic? (maybe this is solved via the max seat changes instead).